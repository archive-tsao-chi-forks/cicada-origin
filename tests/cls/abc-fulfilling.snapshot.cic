def the_nat_type(_: Trivial): Type = Nat

class ABC [ a: Trivial, b: the_nat_type(a), c: String ]

def abc: ABC(sole, 2) = {
  a: sole, b: 2, c: "c"
}

abc

abc.a
abc.b
abc.c
