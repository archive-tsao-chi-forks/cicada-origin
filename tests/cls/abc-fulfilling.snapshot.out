the ABC [
  a: Trivial = sole
  b: Nat = 2
  c: String
] {
  a: sole
  b: 2
  c: "c"
}
the Trivial sole
the Nat 2
the String "c"

